text/0000-post-installation-mirror.md

```diff
@@ -0,0 +1,124 @@
+- users who need to operate within network restricted environments (NRE)
+- users who value deterministic dependency resolution and hermetic builds above convenience
+
+*How should this feature be introduced and taught to existing Yarn users?*
+Explain the intended use case with illustrated work flow.
+
+# Drawbacks
+
+- Somewhat a deviation from current node community norm.  This feature may potentially require additional teaching.
+- Added complexity for installation work flow and offline mirror structure
+- Depending on directory structure to identify post installation modules
+
+The following points are drawbacks under certain circumstances and advantages for other circumstances.
+
+- Native extension support
+If a node module has native extensions, a stored post installation module will not work on platforms different from where the module is created.
+
+Currently, native extensions are compiled during installation. However, since the compiler and libraries are provided by running environment, the compilation output are not guaranteed to be repeatable. To ensure hermetic and repeatable build, a separate RFC is necessary due to the complexity of supporting node native extensions .
+
+- Lost of some flexibility
+In at least one package (cldr-data), [build result can be altered by environment variables $CLDR_COVERAGE](https://github.com/rxaviers/cldr-data-npm/blob/master/install.js#L91).  Caching post installation artifacts will lose this flexibility.
+
+- Update to installation time downloads are ignored / require explicit action
+Installation scripts tends to download the latest version of dependencies.  A stored post installation artifacts will always have the same version of dependencies and thus potentially will not have the latest dependencies.  To update such installation time downloaded dependencies, explicit actions from offline mirror maintainers will be required.
+
+# Alternatives
+
+The following alternatives have been considered:
+
+- Do nothing
+One may argue that this problem is not worthy to be addressed and do nothing is the correct approach. It has been in existence since the inception of npm and the node community has thrived at the same time. A second argument is that users in network restricted environments (NRE) are not the intended customers.
+However, as adoption of node widens, the inability to run node build in network restricted environments has been and will continue to be a hurdle for adoption.  Not addressing this problem is no longer a valid option.
+
+- Work with each individual package owners to make sure package installation is hermetic
+There are several drawbacks of this approach:
+  - Installation scripts may serve legitimate purposes in certain circumstances
+  - Requires significant efforts to educate node module writers
+  - Working on a per package basis and updating all dependent packages might take a long time for the necessary changes to propagate.
+

```

Comment:
This may be a dumb question but why do some npm packages need internet access to be install? Why can't they hold all needed information within the package itself? (aside from defined dependencies)